components/benefit_sponsors/spec/dummy/app/models/benefit_group_assignment.rb

```diff
@@ -129,8 +129,6 @@ def is_case_old?
   end
 
   def plan_year
-    warn "[Deprecated] Instead use benefit application" unless Rails.env.test?
-    return benefit_group.plan_year if is_case_old?
```

Comment:
We'll need the is_case_old? check in the new benefit_application method if we're removing it from this one